services/sam-audio/server.py

```
@@ -55,38 +87,107 @@ def _resolve_device() -> str:
         return "cpu"
 
 
-def _try_load_sam() -> Tuple[Any, Any, str, bool]:
-    """Returns (model, processor, device, mock).
-
-    Real Meta SAM-Audio when `sam_audio` + weights are available (Desktop GPU/CPU).
-    Set SAM_AUDIO_REQUIRE_REAL=1 to fail instead of mock (production Desktop).
-    """
+def _try_load_sam() -> Tuple[Any, Any, str, bool, Optional[str]]:
+    """Returns (model, processor, device, mock, error)."""
     global _sam_bundle
```

Ruff (0.16.1)

[warning] Using the global statement to update `_sam_bundle` is discouraged

(PLW0603)